exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillOperatorTable.java

```diff
@@ -26,34 +33,88 @@
 import org.apache.calcite.sql.SqlOperatorTable;
 import org.apache.calcite.sql.SqlSyntax;
 import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.drill.exec.planner.physical.PlannerSettings;
+import org.apache.drill.exec.server.options.SystemOptionManager;
 
 import java.util.List;
+import java.util.Map;
 
+/**
+ * Implementation of {@link SqlOperatorTable} that contains standard operators and functions provided through
+ * {@link #inner SqlStdOperatorTable}, and Drill User Defined Functions.
+ */
 public class DrillOperatorTable extends SqlStdOperatorTable {
-  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DrillOperatorTable.class);
-
+//  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DrillOperatorTable.class);
   private static final SqlOperatorTable inner = SqlStdOperatorTable.instance();
-  private List<SqlOperator> operators;
-  private ArrayListMultimap<String, SqlOperator> opMap = ArrayListMultimap.create();
+  private final List<SqlOperator> operatorsCalcite = Lists.newArrayList();
+  private final List<SqlOperator> operatorsDefault = Lists.newArrayList();
```

Comment:
We want to safely fall back to original behavior (which does not do inference). So the safest way is to have the ones which do not do inference to have their own operators.